devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -639,128 +707,177 @@ bool YarpRobotLoggerDevice::attachAll(const yarp::dev::PolyDriverList& poly)
         for (const auto& sensorName : m_robotSensorBridge->getTemperatureSensorsList())
         {
-            ok = ok
-                 && m_bufferManager.addChannel({"temperatures::" + sensorName,
-                                                {1, 1}, //
-                                                {"temperature"}});
+            std::string fullTemperatureSensorName
+                = TEMPERATURE_NAME + TREE_DELIM + sensorName;
+            ok &= (*this.*initMetadataFunction)(fullTemperatureSensorName, TemperatureNames);
         }
     }
 
-    // resize the temporary vectors
-    m_jointSensorBuffer.resize(dofs);
+    std::string signalName = "";
+    std::string rtSignalName = "";
 
-    // open the TextLogging port
-    ok = ok && m_textLoggingPort.open(m_textLoggingPortName);
-    // run the thread
-    m_lookForNewLogsThread = std::thread([this] { this->lookForNewLogs(); });
+    for (auto& [name, signal] : m_vectorsCollectionSignals)
+    {
+        std::lock_guard<std::mutex> lock(signal.mutex);
+        BipedalLocomotion::YarpUtilities::VectorsCollection* externalSignalCollection
+            = signal.client.readData(false);
+        if (externalSignalCollection != nullptr)
+        {
+            if (!signal.dataArrived)
+            {
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
+                {
+                    signalName = signal.signalName + TREE_DELIM + key;
+                    const auto& metadata = signal.metadata.vectors.find(key);
+                    if (metadata == signal.metadata.vectors.cend())
+                    {
+                        log()->warn("{} Unable to find the metadata for the signal named {}. The "
+                                    "default one will be used.",
+                                    logPrefix,
+                                    signalName);
+                        (*this.*initMetadataFunction)(signalName, {});
+                    } else
+                    {
+                        (*this.*initMetadataFunction)(signalName, {metadata->second});
+                    }
+                }
+                signal.dataArrived = true;
+            }
+        }
+    }
 
-    // run the thread for reading the exogenous signals
-    m_lookForNewExogenousSignalThread = std::thread([this] { this->lookForExogenousSignals(); });
+    for (auto& [name, signal] : m_vectorSignals)
+    {
+        std::lock_guard<std::mutex> lock(signal.mutex);
+        yarp::sig::Vector* collection = signal.port.read(false);
+        if (collection != nullptr)
+        {
+            if (!signal.dataArrived)
+            {
+                (*this.*initMetadataFunction)(signalName, {});
+                signal.dataArrived = true;
+            }
+        }
+    }
+
+    m_vectorCollectionRTDataServer.populateMetadata(ROBOT_RT_ROOT_NAME + TREE_DELIM + "newMetadata", {"newMetadata"});
+
+    if(sendDataRT)
+        m_vectorCollectionRTDataServer.finalizeMetadata();
```

Comment:
```suggestion
    if(sendDataRT)
    {
        m_vectorCollectionRTDataServer.finalizeMetadata();
    }
```

Reply:
Fixed in [cf0370b](https://github.com/ami-iit/bipedal-locomotion-framework/commit/cf0370bd7a167d2ce02711edb93411a13482fe64)